### PR TITLE
PhoneticManager_ExpandPhonetics: Limit lenght of term to expand - MOD 5767

### PR DESCRIPTION
**Pull request by nafraf**
**Describe the changes in the pull request**

Currently, the function `PhoneticManager_ExpandPhonetics()` does not validate the maximum lenght of the term to expand.  When a user input a massive huge length, the stack-based buffer will end-up with take all stack memory space and trigger a DOS.

Changes done:
1. The max length was defined `MAX_TERM_TO_EXPAND_SIZE 128`, it has the same value that `MAX_NORMALIZE_SIZE`.
2. If the argument len of the function `PhoneticManager_ExpandPhonetics()` is greater than the allowed value.

**Which issues this PR fixes**
1.   [3836](https://github.com/RediSearch/RediSearch/issues/3836)
2. MOD-5767

**Main objects this PR modified**
1. Function `PhoneticManager_ExpandPhonetics()`
3. Function `simpleTokenizer_Next()`

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/phonetic_manager.c [hunk 1]
```diff
@@ -22,6 +24,10 @@ static void PhoneticManager_AddPrefix(char** phoneticTerm) {
 
 void PhoneticManager_ExpandPhonetics(PhoneticManagerCtx* ctx, const char* term, size_t len,
                                      char** primary, char** secondary) {
+  if(len > MAX_TERM_TO_EXPAND_SIZE) {
```
Comment (DvirDukhan) [inline src/phonetic_manager.c:27]: Should we fallback to heap based buffer in this case?

**Review by DvirDukhan — CHANGES_REQUESTED** [review]
Please use the issue description code snippet and add tests
👍

File: src/phonetic_manager.c (hunk 1 above)
Comment (nafraf) [reply src/phonetic_manager.c]: Yes, you are right. Please review the last change. Thanks.

**Review by DvirDukhan — APPROVED** [review]

File: tests/pytests/test_phonetics.py [hunk 2]
```diff
@@ -106,3 +106,19 @@ def testIssue1313(env):
 
     env.expect('FT.CREATE test1 ON HASH SCHEMA topic TEXT PHONETIC dm:en topic2 TEXT NOINDEX').ok()
     env.expect('FT.SEARCH', 'test1', '@topic:(tmp)=>{$phonetic: true}').equal([0])
+
+def testIssue3836(env):
+    env.expect('FT.CREATE idx schema text TEXT PHONETIC dm:en SORTABLE').ok()
+    env.expect('FT.ADD idx doc1 1.0 fields text morfix').ok()
```
[the remark this answers is not in the document]
Comment (nafraf) [reply tests/pytests/test_phonetics.py]: OK. Change done. Thanks.

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**